homeassistant/components/switch/broadlink.py

```diff
@@ -80,24 +80,23 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         config.get(CONF_MAC).encode().replace(b':', b''))
     switch_type = config.get(CONF_TYPE)
 
-    @asyncio.coroutine
-    def _learn_command(call):
+    async def _learn_command(call):
         """Handle a learn command."""
         try:
-            auth = yield from hass.async_add_job(broadlink_device.auth)
+            auth = await hass.async_add_job(broadlink_device.auth)
         except socket.timeout:
             _LOGGER.error("Failed to connect to device, timeout")
             return
         if not auth:
             _LOGGER.error("Failed to connect to device")
             return
 
-        yield from hass.async_add_job(broadlink_device.enter_learning)
+        await hass.async_add_job(broadlink_device.enter_learning)
```

Comment:
In the future, all calls to `async_add_job` should be rewritten as either `async_add_executor_job` or `async_create_task` (if it's a coro)

Reply:
My focus for these PRs was to replace the old `asyncio` syntax (mostly through search and replace with regex). However I'll keep that in mind for future PRs.

Reply:
I know. But since you were doing massive regex operations, I thought I suggest a few more 😝